src/pybind/mgr/balancer/module.py

```diff
@@ -31,7 +31,9 @@ def __init__(self, osdmap, pg_dump, desc=''):
         self.pg_stat = {
             i['pgid']: i['stat_sum'] for i in pg_dump.get('pg_stats', [])
         }
-        self.poolids = [p['pool'] for p in self.osdmap_dump.get('pools', [])]
+        osd_poolids = [p['pool'] for p in self.osdmap_dump.get('pools', [])]
+        pg_poolids = [p['poolid'] for p in pg_dump.get('pool_stats', [])]
+        self.poolids = [p for p in osd_poolids if p in pg_poolids]
```

Comment:
Would be a bit neater (and perhaps more efficient) to do a `set(osd_poolids) & set(pg_poolids)`

Reply:
This is still a work-in-progress and I have some other bugs to hunt :joy: 
Will fix in next version \o/